src/pal/tests/palsuite/file_io/GetFileSizeEx/test1/GetFileSizeEx.c

```diff
@@ -0,0 +1,214 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+/*=====================================================================
+**
+** Source:  GetFileSizeEx.c (test 1)
+**
+** Purpose: Tests the PAL implementation of the GetFileSizeEx function.
+**
+**
+**===================================================================*/
+
+#include <palsuite.h>
+
+const char* szTextFile = "text.txt";
+
+void CleanUp(HANDLE hFile)
+{
+    if (CloseHandle(hFile) != TRUE)
+    {
+        Fail("GetFileSizeEx: ERROR -> Unable to close file \"%s\".\n", 
+            szTextFile);
+    }
+    if (!DeleteFileA(szTextFile))
+    {
+        Fail("GetFileSizeEx: ERROR -> Unable to delete file \"%s\".\n", 
+            szTextFile);
+    }
+}
+
+
+int __cdecl main(int argc, char *argv[])
+{
+    HANDLE hFile = NULL;
+    BOOL bRc = FALSE;
+    DWORD dwRc = 0;
+    DWORD dwError = 0;
+    DWORD dwHighOrder = 0;
+    DWORD dwOffset = 0;
+    DWORD dwReturnedHighOrder = 0;
+    DWORD dwReturnedOffset = 0;
+    DWORD lpNumberOfBytesWritten;
+    LARGE_INTEGER qwFileSize;
+    LARGE_INTEGER qwFileSize2;
+    char * data = "1234567890";
+
+    qwFileSize.QuadPart = 0;
+    qwFileSize2.QuadPart = 0;
+
+    if (0 != PAL_Initialize(argc,argv))
+    {
+        return FAIL;
+    }
+
+
+    /* test on a null file handle */
+    bRc = GetFileSizeEx(hFile, NULL);
```

Comment:
Having implementing this and again testing against Windows, natively there appear to be no checks and it tries to write to the NULL pointer. The proper approach is still to add this?